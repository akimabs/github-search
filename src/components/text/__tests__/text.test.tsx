/**
 * @format
 */

import 'react-native';
import React from 'react';
import { Text } from '../index';

jest.mock('app/utils');

jest.useFakeTimers();

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('Text', () => {
  describe('Rendering', () => {
    it('should match to snapshot - Text', () => {
      const TextWithSizeCaption = renderer.create(
        <Text type="bold" size={'caption'}>
          Test text
        </Text>,
      );
      const TextWithSizeDescription = renderer.create(
        <Text type="bold" size={'description'}>
          Test text
        </Text>,
      );
      const TextWithSizeTitle = renderer.create(
        <Text type="bold" size={'title'}>
          Test text
        </Text>,
      );
      const TextWithSizeLargeTitle = renderer.create(
        <Text type="bold" size={'large_title'}>
          Test text
        </Text>,
      );
      const TextWithSizeSubtitle = renderer.create(
        <Text type="bold" size={'subtitle'}>
          Test text
        </Text>,
      );
      const TextWithStyle = renderer.create(
        <Text type="bold" size={'subtitle'} style={{ color: 'red' }}>
          Test text
        </Text>,
      );

      expect({
        TextWithSizeCaption,
        TextWithSizeDescription,
        TextWithSizeTitle,
        TextWithSizeLargeTitle,
        TextWithSizeSubtitle,
        TextWithStyle,
      }).toMatchSnapshot('Text snapshot');
    });
  });
});
