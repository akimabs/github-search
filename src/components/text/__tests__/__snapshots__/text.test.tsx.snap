// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Rendering should match to snapshot - Text: Text snapshot 1`] = `
Object {
  "TextWithSizeCaption": <Text
    maxFontSizeMultiplier={1}
    minimumFontScale={12}
    style={
      Object {
        "fontSize": 27,
        "textAlign": "left",
      }
    }
  >
    Test text
  </Text>,
  "TextWithSizeDescription": <Text
    maxFontSizeMultiplier={1}
    minimumFontScale={12}
    style={
      Object {
        "fontSize": 24,
        "textAlign": "left",
      }
    }
  >
    Test text
  </Text>,
  "TextWithSizeLargeTitle": <Text
    maxFontSizeMultiplier={1}
    minimumFontScale={12}
    style={
      Object {
        "fontSize": 47,
        "textAlign": "left",
      }
    }
  >
    Test text
  </Text>,
  "TextWithSizeSubtitle": <Text
    maxFontSizeMultiplier={1}
    minimumFontScale={12}
    style={
      Object {
        "fontSize": 31,
        "textAlign": "left",
      }
    }
  >
    Test text
  </Text>,
  "TextWithSizeTitle": <Text
    maxFontSizeMultiplier={1}
    minimumFontScale={12}
    style={
      Object {
        "fontSize": 39,
        "textAlign": "left",
      }
    }
  >
    Test text
  </Text>,
  "TextWithStyle": <Text
    maxFontSizeMultiplier={1}
    minimumFontScale={12}
    style={
      Object {
        "color": "red",
        "fontSize": 31,
        "textAlign": "left",
      }
    }
  >
    Test text
  </Text>,
}
`;
